pkg/asset/agent/joiner/addnodesconfig.go

```diff
@@ -35,7 +35,8 @@ type Config struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Hosts []agent.Host `json:"hosts,omitempty"`
+	CPUArchitecture string       `json:"cpuArchitecture,omitempty"`
```

Comment:
Being an host specific attribute,  I think it'd sound more natural to specify it for each host?

Reply:
Doing so would require generating an ISO for each host since each host may have a different architecture. I think having architecture at node-config level is ok, because the ISO is generated for a particular architecture.

Reply:
Yeah, I think it makes sense. Also, it does not seem to be a realistic case the scenario where multiple nodes with different arch will be added at the same time to the target cluster

Reply:
Btw, by reviewing the naming, I think that `Architecture` could be more in sync with the default field name also used in [install-config.yaml](https://github.com/openshift/installer/blob/b2876fc66fb8292443a3ff19d6db2d9d829e09f2/data/data/install.openshift.io_installconfigs.yaml#L107)